pkg/arvo/app/http-server-store.hoon

```diff
@@ -0,0 +1,220 @@
+      :-  [%pass url-base %arvo %e %connect [~ url-base] %http-server-store]~
+      this(serving (~(put by serving) url-base [clay-base.act public.act]))
+    ::
+        %unserve-dir
+      :-  [%pass url-base.act %arvo %e %disconnect [~ url-base.act]]~
+      this(serving (~(del by serving) url-base.act))
+    ::
+        %toggle-permission
+      ?.  (~(has by serving) url-base.act)
+        ~|("url is not bound" !!)
+      =/  [clay-base=path public=?]  (~(got by serving) url-base.act)
+      :-  ~
+      this(serving (~(put by serving) url-base.act [clay-base !public]))
+    ::
+        %set-landscape-homepage-prefix
+      =.  landscape-homepage-prefix.configuration  prefix.act
+      :_  this
+      (give [%configuration configuration])
+    ==
+    ::
+    ++  give
+      |=  =update
+      ^-  (list card)
+      [%give %fact [/all]~ [%http-server-update !>(update)]]~
+    --
+  ::
+  ++  handle-http-request
+    |=  =inbound-request:eyre
+    ^-  simple-payload:http
+    |^
+    =*  req       request.inbound-request
+    =*  headers   header-list.req
+    =/  req-line  (parse-request-line url.req)
+    ?.  =(method.req %'GET')  not-found:gen
+    =?  req-line  ?=(~ ext.req-line)
+      [[[~ %html] ~['index']] args.req-line]
+    ?>  ?=(^ ext.req-line)
+    ?~  site.req-line
+      not-found:gen
+    =*  url-prefix  landscape-homepage-prefix.configuration
+    ?.  ?|  ?&(?=(^ url-prefix) =((need url-prefix) i.site.req-line))
+            =(url-prefix ~)
+        ==
+      not-found:gen
+    ::
+    ?:  ?=([%'~landscape' %js %session ~] site.req-line)
+      %+  require-authorization-simple:app
+        inbound-request
+      %-  js-response:gen
+      (as-octt:mimes:html "window.ship = '{+:(scow %p our.bowl)}';")
+    ::
+    =/  [payload=simple-payload:http public=?]  (get-file req-line)
+    ?:  public  payload
+    (require-authorization-simple:app inbound-request payload)
+    ::
+    ++  get-file
+      |=  req-line=request-line
+      ^-  [simple-payload:http ?]
+      =/  pax=path  (snoc site.req-line (need ext.req-line))
+      =/  clay-path=(unit [path ?])  (get-clay-path pax)
+      ?~  clay-path  [not-found:gen %.n]
+      =/  scry-path
+        :*  (scot %p our.bowl)
+            q.byk.bowl
+            (scot %da now.bowl)
+            -.u.clay-path
+        ==
+      ?.  .^(? %cu scry-path)  [not-found:gen %.n]
+      =/  file  (as-octs:mimes:html .^(@ %cx scry-path))
+      :_  +.u.clay-path
+      ?+  ext.req-line  not-found:gen
+          [~ %html]  (html-response:gen file)
+          [~ %js]    (js-response:gen file)
+          [~ %css]   (css-response:gen file)
+          [~ %png]   (png-response:gen file)
+      ==
+    ::
+    ++  get-clay-path
+      |=  pax=path
+      ^-  (unit [path ?])
+      =/  first-try  (match-clay-path pax (~(del by serving) /))
+      ?^  first-try  first-try
+      =/  root  (~(get by serving) /)
+      ?~  root  ~
+      (match-clay-path pax (~(gas by *^serving) [[/ u.root] ~]))
```

Comment:
This is... special-casing the `/` path, because every possible path is a suffix of it? You could probably rewrite `+match-clay-path` to find the _most specific_ match instead, so you don't have to do two passes here, and can also override with more specific content in other places, where necessary.

Eyre behaves this way too, wrt bindings.

Reply:
This is good as is. It's not doing any extra work with this approach.